src/api/java/com/minecolonies/api/research/registry/IResearchEffectRegistry.java

```diff
@@ -0,0 +1,41 @@
+package com.minecolonies.api.research.registry;
+
+import com.minecolonies.api.IMinecoloniesAPI;
+import com.minecolonies.api.research.effects.IResearchEffect;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ *   An interface providing information about Minecolonies Research Effects,
+ *   and the ability to register new effects.  Also some helper classes to reduce later sorting.
+ *   This ResearchEffectRegistry should include both research effects regardless of status:
+ *   Use {@link com.minecolonies.api.research.effects.IResearchEffectManager}
+ *   to set and apply effects for completed research.
+ *   Use {MinecoloniesAPIProxy.getInstance().getGlobalResearchTree()}
+ *   to add research to the University Research System.
+ */
+
+public interface IResearchEffectRegistry
```

Comment:
I've been wondering wether we need this at all, as our current effects are already very general and the way I see it are enclosing all possible use cases. We've got a boolean yes/no effect and a numeric effect to retrive as information, so not sure if this isnt just an overkill.

Reply:
Raycoms requested it in https://github.com/ldtteam/minecolonies/issues/5842#issuecomment-735360052 . It's implemented as an interface to specific effects (healthboost, invslots, citizen-cap, unlock Building X) rather than of effect types (unlockBuilding, unlockAbility, multiplication, addition).

It's currently not used at all, and the only real use case I can come up with inside the mod proper is for more generic implementations, so if it's out of scope it's not hard to delete.  I think the intended situation was more for outside modders intending to register effects without putting them into the research tree (such as a consumed item from another mod unlocking a crafterrecipe) or where the effect might not normally exist and they can't use the existing blockhut checkResearch approach (such as if they want a behavior in their own mod to be dependent on a research unlock, but only if that unlock exists).

Reply:
Yeah, I had talked that over with someaddons 1-2 days ago. I had thought we might need a bunch of additional ones, but I think overally we're actually fine. The generic types we have (number, unlock) are probably all we need. And even if someone come sup with sth more complex we can still PR that into the mod.